logic/logic.go

```diff
@@ -29,16 +32,58 @@ type operation struct {
 	Group         []string
 }
 
+// Deprecated
 var spec *langSpec
+
+// Deprecated
 var opcodes []operation
 
 // CheckProgram performs basic program validation: instruction count and program cost
+// Deprecated: `langspec.json` can no longer correctly to depicting the cost model (as of 2022.08.22),
+// also to minimize the work in updating SDKs per AVM release, we are deprecating`langspec.json` across all SDKs.
+// The behavior of `CheckProgram` relies on `langspec.json`. Thus, this method is being deprecated.
 func CheckProgram(program []byte, args [][]byte) error {
 	_, _, err := ReadProgram(program, args)
 	return err
 }
 
+func isAsciiPrintableByte(symbol byte) bool {
+	isBreakLine := symbol == '\n'
+	isStdPrintable := symbol >= ' ' && symbol <= '~'
+	return isBreakLine || isStdPrintable
+}
+
+func isAsciiPrintable(program []byte) bool {
+	for _, b := range program {
+		if !isAsciiPrintableByte(b) {
+			return false
+		}
+	}
+	return true
+}
+
+// SanityCheckProgram performs heuristic program validation:
+// check if passed in bytes are Algorand address or is B64 encoded, rather than Teal bytes
+func SanityCheckProgram(program []byte) error {
```

Comment:
@ahangsu Let me know what you think of the following proposal?

I propose marking `SanityCheckProgram` (and analogous methods in other SDKs) private.

Since the check is _only_ a set of heuristics, my thinking is that we should _not_ encourage API users to rely on the check.

Reply:
good point, I made it private in js-algorand-sdk, but LogicsigSignature in `crypto` package cannot get access to this heuristic method. Is there a preferred way to make this private, while still can be accessed in other package?

Reply:
I made the Js/Py-sdk's `sanityCheckProgram` private or protected, but in Java/Go, it does not seem straightforward to both hide this from user and export it internally

Reply:
@ahangsu Here's my thinking:

For Java, provided we agree to mark `sanityCheckProgram` private, then I suggest:
* Move definition of `sanityCheckProgram` into `LogicsigSignature`.
* Mark `public class Logic` as deprecated because it'll contain no non-deprecated functionality. 
  * Implies we can remove deprecations on specific fields + methods because the class is deprecated.

For Go, I offer the same suggestion + caveats with `SanityCheckProgram` being moved into `crypto.go`.

Reply:
that sounds reasonable to me, working on it now.

Reply:
I've only done a cursory review so far, but I'll leave my thoughts here.

It seems like we cannot avoid changing the behavior of the SDKs in some way. Indeed, that's the goal of this PR. That being the case, I'd opt to change the existing `CheckProgram` function to become `SanityCheckProgram`. That way, we do not export any new functions and we can keep the implementation in this package.

Reply:
for function exporting issue, if we move heuristic sanity check function into `LogicsigSignature`, and make them private methods, then we are just simply adding an internal method, rather than something exported. We are deprecating `CheckProgram` anyways, I think...

Reply:
Ok, I will not push further. The latest changes make sense